tests/unit_tests/client/address_book/storage.cc

```diff
@@ -0,0 +1,196 @@
+ *                                                                                            //
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be       //
+ *    used to endorse or promote products derived from this software without specific         //
+ *    prior written permission.                                                               //
+ *                                                                                            //
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY        //
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF    //
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL     //
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,       //
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,               //
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS    //
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,          //
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF    //
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.               //
+ */
+
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+
+#include <iostream>
+
+#include "client/address_book/storage.h"
+
+#include "core/router/context.h"
+
+struct AddressBookStorageFixture
+{
+  /// @alias Subscription
+  /// @brief Subscription alias for readability and convenience
+  using Subscription = kovri::client::AddressBook::Subscription;
+
+  AddressBookStorageFixture() : m_Storage(nullptr)
+  {
+    // Setup temporary data directory
+    kovri::core::context.SetCustomDataDir(temp_path.string());
+
+    // Ensure Client and AddressBook paths exist
+    kovri::core::EnsurePath(
+        kovri::core::GetPath(kovri::core::Path::Client));
+
+    kovri::core::EnsurePath(
+        kovri::core::GetPath(kovri::core::Path::AddressBook));
+
+    // Setup storage
+    m_Storage = GetStorageInstance();
+  }
+
+  ~AddressBookStorageFixture()
+  {
+    RemoveFiles();
+  }
+
+  /// @brief Remove files created by AddressBookStorage
+  void RemoveFiles() try
+    {
+      boost::filesystem::directory_iterator end_itr;
+      for (boost::filesystem::directory_iterator dir_itr(
+               kovri::core::GetPath(kovri::core::Path::AddressBook));
+           dir_itr != end_itr;
+           ++dir_itr)
+        {
+          auto temp_file = dir_itr->path().filename();
+
+          // Only remove files created by AddressBookStorage
+          if (!boost::starts_with(temp_file.string(), test_prefix))
+            boost::filesystem::remove(temp_file);
+        }
+    }
+  catch (boost::filesystem::filesystem_error& er)
```

Comment:
We have an exception dispatcher. If you want to add filesystem specifics, go ahead.